include/fixed_containers/enum_utils.hpp

```diff
@@ -364,14 +365,59 @@ class SkeletalRichEnumStorageBase<RichEnumType,
 
 }  // namespace fixed_containers::rich_enums_detail
 
+template <typename RichEnumType>
+class RichEnumConstantProxy
```

Comment:
I generally approve of this idea. I am leaving my thoughts as a comment rather than as PR review such that replies are simpler.

Specifically, I like that this change makes the common use cases of `RichEnums` simpler. For example, I like that we can omit `operator()` when comparing enums such as:
```cpp
TestRichEnum1::value_of(BE::C_ONE) == TestRichEnum1::C_ONE
```
or in switch statements such as:
```cpp
case TestRichEnum1::C_ONE:
```

However, the current approach has some downsides in my eyes.

1. The first downside in my opinion is that calling member functions of `RichEnums` now requires `operator->`, which in my opinion is an operator which conveys pointer like semantics which imply nullptr possibility. However, since the pointer will never be null in this case, this is a purely aesthetic concern and unfortunately I don't currently have alternate suggestions. Example:
```cpp
TestRichEnum1::C_FOUR->ordinal()
```
2. I am not in favor of the
```cpp
// TRANSITION
constexpr const RichEnumType& operator()()
```
In my opinion, this will result in having two different ways of doing the same thing and I expect both to stick around. From what I've seen, things in `transition::` tend to stick around ;) . I suggest that we make this a breaking change that requires users to change their code. That is unless this repository has some convention of backwards compatibility.